### PR TITLE
[4.6] capsys: ensure fd is unbuffered

### PR DESCRIPTION
**Pull request by blueyed**
Fixes https://github.com/pytest-dev/pytest/issues/5134.

### CODE REVIEW

**Comment by nicoddemus** [comment]
Tests are failing, regardless, @antocuni any chance of testing this branch for your case and see if it fixes it?

File: src/_pytest/capture.py [hunk 1]
```diff
@@ -685,11 +685,25 @@ def done(self):
...
+            and getattr(sys, "__{}__".format(self.name)) is self._old
+            and not hasattr(self, "_fdopen_done")
+            and not sys.platform.startswith("win32")
+        ):
+            self._fdopen_done = True
+            try:
+                # Ensure fd is unbuffered (#5134).
+                self._old = os.fdopen(self._old.fileno(), "wb+", 0)
+            except (UnsupportedOperation, OSError):
+                pass
```
Comment (blueyed) [inline src/_pytest/capture.py]: Note that `_py36_windowsconsoleio_workaround` does something similar already: https://github.com/pytest-dev/pytest/blob/f7ecd7243ccf708008a69b8156a0eeadaad2dad6/src/_pytest/capture.py#L804-L816

**Comment by blueyed** [comment]
I do not really feel like we should have it, since it is not required for py3 anyway, and might break other things.

**Comment by blueyed** [comment]
But FWIW tests are passing by now.

**Comment by blueyed** [comment]
I do not expect any feedback from @antocuni here (chatted with regard to the workaround he added to pyrepl).  Might re-open when I look into that (https://github.com/pypy/pyrepl/pull/22, https://github.com/pypy/pyrepl/commit/3ff3cfaa4e79fed9fe1b3c3a2c56ca46ccbd04dc).

**Pull request closed without merging**